core/src/main/java/org/opensearch/sql/analysis/Analyzer.java

```diff
@@ -469,8 +470,13 @@ public LogicalPlan visitSort(Sort node, AnalysisContext context) {
         node.getSortList().stream()
             .map(
                 sortField -> {
-                  Expression expression = optimizer.optimize(
-                      expressionAnalyzer.analyze(sortField.getField(), context), context);
+                  var analyzed = expressionAnalyzer.analyze(sortField.getField(), context);
+                  if (analyzed == null) {
+                    throw new UnsupportedOperationException(
+                        String.format("Invalid use of expression %s", sortField.getField())
```

Comment:
It results to the following message when I run an incorrect query ([example](https://github.com/Bit-Quill/opensearch-project-sql/pull/278#issuecomment-1599275236))
```json
{
  "error":
  {
    "reason": "There was internal problem at backend",
    "details": "Invalid use of expression NestedAllTupleFields(path=message)",
    "type": "UnsupportedOperationException"
  }
}
```
It is possible to make it more user-friendly by updating `toString` method in `NestedAllTupleFields`?